python/tests/test_umap.py

```diff
@@ -375,3 +364,47 @@ def assert_umap_model(model: UMAPModel) -> None:
         trust_diff = loc_umap - dist_umap
 
         assert trust_diff <= 0.15
+
+
+def test_umap_sample_fraction(gpu_number: int) -> None:
+    from cuml.datasets import make_blobs
+
+    X, _ = make_blobs(
+        5000,
+        3000,
+        centers=42,
+        cluster_std=0.1,
+        dtype=np.float32,
+        random_state=10,
+    )
+
+    with CleanSparkSession() as spark:
+        pyspark_type = "float"
+        feature_cols = [f"c{i}" for i in range(X.shape[1])]
+        schema = [f"{c} {pyspark_type}" for c in feature_cols]
+        df = spark.createDataFrame(X.tolist(), ",".join(schema))
+        df = df.withColumn("features", array(*feature_cols)).drop(*feature_cols)
+
+        sample_fraction = 0.5
+        umap = (
+            UMAP(num_workers=gpu_number)
```

Comment:
Can random_state be set to make it deterministic?  Otherwise, seems will have same failure rate as current test.

Reply:
thanks for catching that. fixed